lib/reader/src/parser.rs

```diff
@@ -158,10 +162,31 @@ impl<'a> Context<'a> {
         self.map.def_ebb(src_ebb, ebb, loc).and(Ok(ebb))
     }
 
+    fn add_alias(&mut self, src: Value, dest: Value, loc: &Location) -> Result<()> {
+        if self.aliases.insert(src, dest).is_some() {
+            err!(loc, "duplicate alias: {} -> {}", src, dest)
+        } else {
+            Ok(())
+        }
+    }
+
     // The parser creates all instructions with Ebb and Value references using the source file
     // numbering. These references need to be rewritten after parsing is complete since forward
     // references are allowed.
     fn rewrite_references(&mut self) -> Result<()> {
+        for (&source_from, &source_to) in &self.aliases {
+            let ir_to = self.map
+                .get_value(source_to)
+                .expect(&*format!("IR destination value alias not found for {}", source_to));
```

Comment:
I see what you mean.

Source locations for aliases are a bit tricky since the same alias can be defined in many places (as long as it's consistent). I think a source location that simply represents the first time you saw the alias would be fine, though.

Storing the first-seen source locations in the hash map makes sense to me.